<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>SAGA C++ API: saga::exception Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<script type="text/javascript">
function hasClass(ele,cls) {
  return ele.className.match(new RegExp('(\\s|^)'+cls+'(\\s|$)'));
}

function addClass(ele,cls) {
  if (!this.hasClass(ele,cls)) ele.className += " "+cls;
}

function removeClass(ele,cls) {
  if (hasClass(ele,cls)) {
    var reg = new RegExp('(\\s|^)'+cls+'(\\s|$)');
    ele.className=ele.className.replace(reg,' ');
  }
}

function toggleVisibility(linkObj) {
 var base = linkObj.getAttribute('id');
 var summary = document.getElementById(base + '-summary');
 var content = document.getElementById(base + '-content');
 var trigger = document.getElementById(base + '-trigger');
 if ( hasClass(linkObj,'closed') ) {
   summary.style.display = 'none';
   content.style.display = 'block';
   trigger.src = 'open.png';
   removeClass(linkObj,'closed');
   addClass(linkObj,'opened');
 } else if ( hasClass(linkObj,'opened') ) {
   summary.style.display = 'block';
   content.style.display = 'none';
   trigger.src = 'closed.png';
   removeClass(linkObj,'opened');
   addClass(linkObj,'closed');
 }
 return false;
}
</script>
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="main.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
    </ul>
  </div>
  <div class="tabs2">
    <ul class="tablist">
      <li><a href="annotated.html"><span>Class&nbsp;List</span></a></li>
      <li><a href="classes.html"><span>Class&nbsp;Index</span></a></li>
      <li><a href="hierarchy.html"><span>Class&nbsp;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&nbsp;Members</span></a></li>
    </ul>
  </div>
  <div class="navpath">
    <ul>
      <li><a class="el" href="namespacesaga.html">saga</a>      </li>
      <li><a class="el" href="classsaga_1_1exception.html">saga::exception</a>      </li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="summary">
<a href="#pub-methods">Public Member Functions</a> &#124;
<a href="#pub-attribs">Public Attributes</a>  </div>
  <div class="headertitle">
<h1>saga::exception Class Reference</h1>  </div>
</div>
<div class="contents">
<!-- doxytag: class="saga::exception" --><!-- doxytag: inherits="std::exception" -->
<p><code>#include &lt;<a class="el" href="exception_8hpp_source.html">exception.hpp</a>&gt;</code></p>
<div id="dynsection-0" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-0-trigger" src="closed.png"/> Inheritance diagram for saga::exception:</div>
<div id="dynsection-0-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-0-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classsaga_1_1exception__inherit__graph.png" border="0" usemap="#saga_1_1exception_inherit__map" alt="Inheritance graph"/></div>
<map name="saga_1_1exception_inherit__map" id="saga_1_1exception_inherit__map">
<area shape="rect" id="node5" href="classsaga_1_1no__success.html" title="{saga::no_success\n||+ no_success()\l+ no_success()\l+ no_success()\l}" alt="" coords="4,432,136,560"/><area shape="rect" id="node7" href="classsaga_1_1not__implemented.html" title="{saga::not_implemented\n||+ not_implemented()\l+ not_implemented()\l+ not_implemented()\l}" alt="" coords="160,432,332,560"/><area shape="rect" id="node9" href="classsaga_1_1parameter__exception.html" title="{saga::parameter_exception\n||# parameter_exception()\l# parameter_exception()\l# parameter_exception()\l}" alt="" coords="357,432,552,560"/><area shape="rect" id="node15" href="classsaga_1_1security__exception.html" title="{saga::security_exception\n||# security_exception()\l# security_exception()\l# security_exception()\l}" alt="" coords="626,432,805,560"/><area shape="rect" id="node23" href="classsaga_1_1state__exception.html" title="{saga::state_exception\n||# state_exception()\l# state_exception()\l# state_exception()\l}" alt="" coords="1310,432,1471,560"/><area shape="rect" id="node2" title="STL class." alt="" coords="398,6,511,96"/><area shape="rect" id="node11" href="classsaga_1_1bad__parameter.html" title="{saga::bad_parameter\n||+ bad_parameter()\l+ bad_parameter()\l+ bad_parameter()\l}" alt="" coords="154,608,309,736"/><area shape="rect" id="node13" href="classsaga_1_1incorrect__url.html" title="{saga::incorrect_url\n||+ incorrect_url()\l+ incorrect_url()\l+ incorrect_url()\l}" alt="" coords="335,608,475,736"/><area shape="rect" id="node17" href="classsaga_1_1authentication__failed.html" title="{saga::authentication_failed\n||+ authentication_failed()\l+ authentication_failed()\l+ authentication_failed()\l}" alt="" coords="500,608,695,736"/><area shape="rect" id="node19" href="classsaga_1_1authorization__failed.html" title="{saga::authorization_failed\n||+ authorization_failed()\l+ authorization_failed()\l+ authorization_failed()\l}" alt="" coords="719,608,907,736"/><area shape="rect" id="node21" href="classsaga_1_1permission__denied.html" title="{saga::permission_denied\n||+ permission_denied()\l+ permission_denied()\l+ permission_denied()\l}" alt="" coords="932,608,1113,736"/><area shape="rect" id="node25" href="classsaga_1_1already__exists.html" title="{saga::already_exists\n||+ already_exists()\l+ already_exists()\l+ already_exists()\l}" alt="" coords="1138,608,1288,736"/><area shape="rect" id="node27" href="classsaga_1_1does__not__exist.html" title="{saga::does_not_exist\n||+ does_not_exist()\l+ does_not_exist()\l+ does_not_exist()\l}" alt="" coords="1312,608,1468,736"/><area shape="rect" id="node29" href="classsaga_1_1incorrect__state.html" title="{saga::incorrect_state\n||+ incorrect_state()\l+ incorrect_state()\l+ incorrect_state()\l}" alt="" coords="1493,608,1648,736"/><area shape="rect" id="node31" href="classsaga_1_1timeout.html" title="{saga::timeout\n||+ timeout()\l+ timeout()\l+ timeout()\l}" alt="" coords="1672,608,1781,736"/></map>
<center><span class="legend">[<a target="top" href="graph_legend.html">legend</a>]</span></center></div>

<p><a href="classsaga_1_1exception-members.html">List of all members.</a></p>
<table class="memberdecls">
<tr><td colspan="2"><h2><a name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classsaga_1_1exception.html#a492a5639c5096cf4ff7c7f114642fbb3">~exception</a> ()  throw ()</td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">Destroys the exception.  <a href="#a492a5639c5096cf4ff7c7f114642fbb3"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">char const *&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classsaga_1_1exception.html#a3e2d492694a712a210552e7d783744c5">what</a> () const   throw ()</td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">Brief description starts here.  <a href="#a3e2d492694a712a210552e7d783744c5"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">char const *&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classsaga_1_1exception.html#a1ea02cb0e7d7040522efebbb33f12d4c">get_message</a> () const   throw ()</td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">Gets the message associated with the exception.  <a href="#a1ea02cb0e7d7040522efebbb33f12d4c"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="el" href="namespacesaga.html#a6ca051bb87f4c5c63d80b2bc1502b377">saga::error</a>&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classsaga_1_1exception.html#a8899bf9c71a7e8cb0665a9a8c257bab3">get_error</a> () const   throw ()</td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">Brief description starts here.  <a href="#a8899bf9c71a7e8cb0665a9a8c257bab3"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="el" href="classsaga_1_1object.html">saga::object</a>&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classsaga_1_1exception.html#a9e29d7af37e2f2066015680bb68ede43">get_object</a> () const   throw ()</td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">Gets the SAGA object associated with exception.  <a href="#a9e29d7af37e2f2066015680bb68ede43"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">std::vector&lt; <a class="el" href="classsaga_1_1exception.html">saga::exception</a> &gt;<br class="typebreak"/>
 const &amp;&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classsaga_1_1exception.html#a956852c277d4c0372e95fceba582870a">get_all_exceptions</a> () const   throw ()</td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">return list of all exceptions encapsulated in this instance  <a href="#a956852c277d4c0372e95fceba582870a"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">std::vector&lt; std::string &gt;&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classsaga_1_1exception.html#a8c28999fb6a1634dd9f24f7d42b093cd">get_all_messages</a> () const </td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">return list of all messages  <a href="#a8c28999fb6a1634dd9f24f7d42b093cd"></a><br/></td></tr>
<tr><td colspan="2"><h2><a name="pub-attribs"></a>
Public Attributes</h2></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="el" href="classsaga_1_1exception.html">exception</a> std::string &amp;&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classsaga_1_1exception.html#a33ed9d3aadf5dacc35c7558b90964545">m</a></td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">Creates the exception.  <a href="#a33ed9d3aadf5dacc35c7558b90964545"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="el" href="classsaga_1_1exception.html">exception</a> std::string <a class="el" href="namespacesaga.html#a6ca051bb87f4c5c63d80b2bc1502b377">error</a>&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classsaga_1_1exception.html#a3c1fdc5c6e24fdc40c199444da4dd72e">e</a> = NoSuccess)</td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="el" href="classsaga_1_1exception.html">exception</a> std::vector<br class="typebreak"/>
&lt; <a class="el" href="classsaga_1_1exception.html">saga::exception</a> &gt; &amp;&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classsaga_1_1exception.html#a1c2e41091ed171a19229662c7fa273d3">l</a></td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">Creates the exception, without associating a SAGA object instance.  <a href="#a1c2e41091ed171a19229662c7fa273d3"></a><br/></td></tr>
</table>
<hr/><a name="_details"></a><h2>Detailed Description</h2>
<p>We introduce a exception hierarchy here, which maps onto the exceptions as defined by the SAGA specification:</p>
<p><a class="el" href="classsaga_1_1exception.html">saga::exception</a><br/>
 &nbsp;<a class="el" href="classsaga_1_1not__implemented.html">saga::not_implemented</a><br/>
 &nbsp;<a class="el" href="classsaga_1_1parameter__exception.html">saga::parameter_exception</a><br/>
 &nbsp;&nbsp;<a class="el" href="classsaga_1_1incorrect__url.html">saga::incorrect_url</a><br/>
 &nbsp;&nbsp;<a class="el" href="classsaga_1_1bad__parameter.html">saga::bad_parameter</a><br/>
 &nbsp;<a class="el" href="classsaga_1_1state__exception.html">saga::state_exception</a><br/>
 &nbsp;&nbsp;<a class="el" href="classsaga_1_1already__exists.html">saga::already_exists</a><br/>
 &nbsp;&nbsp;<a class="el" href="classsaga_1_1does__not__exist.html">saga::does_not_exist</a><br/>
 &nbsp;&nbsp;<a class="el" href="classsaga_1_1incorrect__state.html">saga::incorrect_state</a><br/>
 &nbsp;&nbsp;<a class="el" href="classsaga_1_1timeout.html">saga::timeout</a><br/>
 &nbsp;<a class="el" href="classsaga_1_1security__exception.html">saga::security_exception</a><br/>
 &nbsp;&nbsp;<a class="el" href="classsaga_1_1permission__denied.html">saga::permission_denied</a><br/>
 &nbsp;&nbsp;<a class="el" href="classsaga_1_1authorization__failed.html">saga::authorization_failed</a><br/>
 &nbsp;&nbsp;<a class="el" href="classsaga_1_1authentication__failed.html">saga::authentication_failed</a><br/>
 &nbsp;<a class="el" href="classsaga_1_1no__success.html">saga::no_success</a><br/>
 </p>

<p>Definition at line <a class="el" href="exception_8hpp_source.html#l00052">52</a> of file <a class="el" href="exception_8hpp_source.html">exception.hpp</a>.</p>
<hr/><h2>Constructor &amp; Destructor Documentation</h2>
<a class="anchor" id="a492a5639c5096cf4ff7c7f114642fbb3"></a><!-- doxytag: member="saga::exception::~exception" ref="a492a5639c5096cf4ff7c7f114642fbb3" args="()" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">saga::exception::~exception </td>
          <td>(</td>
          <td class="paramname"></td>
          <td>&nbsp;)&nbsp;</td>
          <td>  throw ()<code> [inline]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Destroys the exception. </p>

<p>Definition at line <a class="el" href="exception_8hpp_source.html#l00103">103</a> of file <a class="el" href="exception_8hpp_source.html">exception.hpp</a>.</p>

</div>
</div>
<hr/><h2>Member Function Documentation</h2>
<a class="anchor" id="a3e2d492694a712a210552e7d783744c5"></a><!-- doxytag: member="saga::exception::what" ref="a3e2d492694a712a210552e7d783744c5" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">char const * saga::exception::what </td>
          <td>(</td>
          <td class="paramname"></td>
          <td>&nbsp;)&nbsp;</td>
          <td> const  throw ()</td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Brief description starts here. </p>

<p>Definition at line <a class="el" href="exception_8cpp_source.html#l00072">72</a> of file <a class="el" href="exception_8cpp_source.html">exception.cpp</a>.</p>

</div>
</div>
<a class="anchor" id="a1ea02cb0e7d7040522efebbb33f12d4c"></a><!-- doxytag: member="saga::exception::get_message" ref="a1ea02cb0e7d7040522efebbb33f12d4c" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">char const * saga::exception::get_message </td>
          <td>(</td>
          <td class="paramname"></td>
          <td>&nbsp;)&nbsp;</td>
          <td> const  throw ()</td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Gets the message associated with the exception. </p>
<dl class="return"><dt><b>Returns:</b></dt><dd>The error message. </dd></dl>

<p>Definition at line <a class="el" href="exception_8cpp_source.html#l00081">81</a> of file <a class="el" href="exception_8cpp_source.html">exception.cpp</a>.</p>

</div>
</div>
<a class="anchor" id="a8899bf9c71a7e8cb0665a9a8c257bab3"></a><!-- doxytag: member="saga::exception::get_error" ref="a8899bf9c71a7e8cb0665a9a8c257bab3" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="namespacesaga.html#a6ca051bb87f4c5c63d80b2bc1502b377">saga::error</a> saga::exception::get_error </td>
          <td>(</td>
          <td class="paramname"></td>
          <td>&nbsp;)&nbsp;</td>
          <td> const  throw ()</td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Brief description starts here. </p>

<p>Definition at line <a class="el" href="exception_8cpp_source.html#l00089">89</a> of file <a class="el" href="exception_8cpp_source.html">exception.cpp</a>.</p>

</div>
</div>
<a class="anchor" id="a9e29d7af37e2f2066015680bb68ede43"></a><!-- doxytag: member="saga::exception::get_object" ref="a9e29d7af37e2f2066015680bb68ede43" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classsaga_1_1object.html">saga::object</a> saga::exception::get_object </td>
          <td>(</td>
          <td class="paramname"></td>
          <td>&nbsp;)&nbsp;</td>
          <td> const  throw ()</td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Gets the SAGA object associated with exception. </p>
<dl class="return"><dt><b>Returns:</b></dt><dd>The object associated with the exception. </dd></dl>

<p>Definition at line <a class="el" href="exception_8cpp_source.html#l00098">98</a> of file <a class="el" href="exception_8cpp_source.html">exception.cpp</a>.</p>

</div>
</div>
<a class="anchor" id="a956852c277d4c0372e95fceba582870a"></a><!-- doxytag: member="saga::exception::get_all_exceptions" ref="a956852c277d4c0372e95fceba582870a" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">std::vector&lt; <a class="el" href="classsaga_1_1exception.html">saga::exception</a> &gt; const &amp; saga::exception::get_all_exceptions </td>
          <td>(</td>
          <td class="paramname"></td>
          <td>&nbsp;)&nbsp;</td>
          <td> const  throw ()</td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>return list of all exceptions encapsulated in this instance </p>

<p>Definition at line <a class="el" href="exception_8cpp_source.html#l00105">105</a> of file <a class="el" href="exception_8cpp_source.html">exception.cpp</a>.</p>

</div>
</div>
<a class="anchor" id="a8c28999fb6a1634dd9f24f7d42b093cd"></a><!-- doxytag: member="saga::exception::get_all_messages" ref="a8c28999fb6a1634dd9f24f7d42b093cd" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">std::vector&lt; std::string &gt; saga::exception::get_all_messages </td>
          <td>(</td>
          <td class="paramname"></td>
          <td>&nbsp;)&nbsp;</td>
          <td> const</td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>return list of all messages </p>
<dl class="note"><dt><b>Note:</b></dt><dd>may throw std::bad_alloc, use with care </dd></dl>

<p>Definition at line <a class="el" href="exception_8cpp_source.html#l00115">115</a> of file <a class="el" href="exception_8cpp_source.html">exception.cpp</a>.</p>

</div>
</div>
<hr/><h2>Member Data Documentation</h2>
<a class="anchor" id="a33ed9d3aadf5dacc35c7558b90964545"></a><!-- doxytag: member="saga::exception::m" ref="a33ed9d3aadf5dacc35c7558b90964545" args="" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classsaga_1_1exception.html">exception</a> std::string&amp; <a class="el" href="classsaga_1_1exception.html#a33ed9d3aadf5dacc35c7558b90964545">saga::exception::m</a></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Creates the exception. </p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table border="0" cellspacing="2" cellpadding="0">
    <tr><td valign="top"></td><td valign="top"><em>obj</em>&nbsp;</td><td>The object associated with the exception. </td></tr>
    <tr><td valign="top"></td><td valign="top"><em>message</em>&nbsp;</td><td>The message to be associated with the new exception. </td></tr>
  </table>
  </dd>
</dl>
<dl class="return"><dt><b>Returns:</b></dt><dd>The newly created exception.</dd></dl>
<p>Creates the exception. </p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table border="0" cellspacing="2" cellpadding="0">
    <tr><td valign="top"></td><td valign="top"><em>obj</em>&nbsp;</td><td>The object associated with the exception. </td></tr>
    <tr><td valign="top"></td><td valign="top"><em>message</em>&nbsp;</td><td>The message to be associated with the new exception. </td></tr>
  </table>
  </dd>
</dl>
<dl class="return"><dt><b>Returns:</b></dt><dd>The newly created exception.</dd></dl>
<p>Creates the exception. </p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table border="0" cellspacing="2" cellpadding="0">
    <tr><td valign="top"></td><td valign="top"><em>obj</em>&nbsp;</td><td>The object associated with the exception. </td></tr>
    <tr><td valign="top"></td><td valign="top"><em>message</em>&nbsp;</td><td>The message to be associated with the new exception. </td></tr>
    <tr><td valign="top"></td><td valign="top"><em>e</em>&nbsp;</td><td>Use this error code for the generated exception </td></tr>
  </table>
  </dd>
</dl>
<dl class="return"><dt><b>Returns:</b></dt><dd>The newly created exception.</dd></dl>
<p>Creates the exception, without associating a SAGA object instance </p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table border="0" cellspacing="2" cellpadding="0">
    <tr><td valign="top"></td><td valign="top"><em>message</em>&nbsp;</td><td>The message to be associated with the new exception. </td></tr>
  </table>
  </dd>
</dl>
<dl class="return"><dt><b>Returns:</b></dt><dd>The newly created exception. </dd></dl>

<p>Definition at line <a class="el" href="exception_8hpp_source.html#l00092">92</a> of file <a class="el" href="exception_8hpp_source.html">exception.hpp</a>.</p>

</div>
</div>
<a class="anchor" id="a3c1fdc5c6e24fdc40c199444da4dd72e"></a><!-- doxytag: member="saga::exception::e" ref="a3c1fdc5c6e24fdc40c199444da4dd72e" args="" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classsaga_1_1exception.html">exception</a> std::string <a class="el" href="namespacesaga.html#a6ca051bb87f4c5c63d80b2bc1502b377">error</a> <a class="el" href="classsaga_1_1exception.html#a3c1fdc5c6e24fdc40c199444da4dd72e">saga::exception::e</a> = NoSuccess)</td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Definition at line <a class="el" href="exception_8hpp_source.html#l00092">92</a> of file <a class="el" href="exception_8hpp_source.html">exception.hpp</a>.</p>

</div>
</div>
<a class="anchor" id="a1c2e41091ed171a19229662c7fa273d3"></a><!-- doxytag: member="saga::exception::l" ref="a1c2e41091ed171a19229662c7fa273d3" args="" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classsaga_1_1exception.html">exception</a> std::vector&lt; <a class="el" href="classsaga_1_1exception.html">saga::exception</a> &gt;&amp; <a class="el" href="classsaga_1_1exception.html#a1c2e41091ed171a19229662c7fa273d3">saga::exception::l</a></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Creates the exception, without associating a SAGA object instance. </p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table border="0" cellspacing="2" cellpadding="0">
    <tr><td valign="top"></td><td valign="top"><em>message</em>&nbsp;</td><td>The message to be associated with the new exception. </td></tr>
  </table>
  </dd>
</dl>
<dl class="return"><dt><b>Returns:</b></dt><dd>The newly created exception. </dd></dl>

<p>Definition at line <a class="el" href="exception_8hpp_source.html#l00099">99</a> of file <a class="el" href="exception_8hpp_source.html">exception.hpp</a>.</p>

</div>
</div>
<hr/>The documentation for this class was generated from the following files:<ul>
<li><a class="el" href="exception_8hpp_source.html">exception.hpp</a></li>
<li><a class="el" href="exception_8cpp_source.html">exception.cpp</a></li>
</ul>
</div>
<hr class="footer"/><address class="footer"><small>Generated on Mon Jun 25 2012 12:33:33 for SAGA C++ API by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
